src/__swaps__/utils/numbers.ts

```diff
@@ -270,28 +268,65 @@ export const convertBipsToPercentage = (value: BigNumberish, decimals = 2): stri
 /**
  * @desc convert from amount value to display formatted string
  */
-export const convertAmountToNativeDisplay = (
-  value: BigNumberish,
-  nativeCurrency: keyof nativeCurrencyType,
-  buffer?: number,
-  skipDecimals?: boolean
-) => {
+export const convertAmountToNativeDisplay = (value: number | string, nativeCurrency: keyof nativeCurrencyType, useThreshold = false) => {
   const nativeSelected = supportedNativeCurrencies?.[nativeCurrency];
-  const { decimals } = nativeSelected;
-  const display = handleSignificantDecimals(value, decimals, buffer, skipDecimals);
-  if (nativeSelected.alignment === 'left') {
-    return `${nativeSelected.symbol}${display}`;
+  const { alignment, decimals: rawDecimals, symbol } = nativeSelected;
+  const decimals = Math.min(rawDecimals, 6);
+
+  const valueNumber = Number(value);
+  const threshold = decimals < 4 ? 0.01 : 0.0001;
+  let thresholdReached = false;
+
+  if (useThreshold && valueNumber < threshold) {
+    thresholdReached = true;
   }
-  return `${display} ${nativeSelected.symbol}`;
+
+  const nativeValue = thresholdReached
+    ? threshold
+    : valueNumber.toLocaleString('en-US', {
+        useGrouping: true,
+        minimumFractionDigits: decimals,
+        maximumFractionDigits: decimals,
+      });
+
+  const nativeDisplay = `${thresholdReached ? '<' : ''}${alignment === 'left' ? symbol : ''}${nativeValue}${alignment === 'right' ? symbol : ''}`;
+
+  return nativeDisplay;
 };
 
-export const convertAmountToNativeDisplayWithThreshold = (value: BigNumberish, nativeCurrency: keyof nativeCurrencyType) => {
+/**
+ * @desc convert from amount value to display formatted string
+ */
+export const convertAmountToNativeDisplayWorklet = (
```

Comment:
does this converts or just formats?
also I always get confused on what native means, my instinct is that native is the network native gas token like eth, but sometimes we use it as the user preferred currency?

Reply:
just formats. i kept the preexisting name for this util